.github/workflows/xcframework.yml

```diff
@@ -0,0 +1,200 @@
+---
+name: xcframework builder
+on:
+    workflow_dispatch:
+    pull_request:
+    push:
+      tags:
+          - v*.*.*
+      branches:
+        - "main"
+permissions:
+    attestations: write
+    contents: write
+    id-token: write
+jobs:
+    iOS-device:
+        runs-on: macos-latest
+        steps:
+            - uses: maxim-lobanov/setup-xcode@60606e260d2fc5762a71e64e74b2174e8ea3c8bd # v1.6.0
+              with:
+                  xcode-version: latest-stable
+            - uses: actions/checkout@v4
+            - name: xcodegen
+              uses: bryce-b/xcodegen-action@patch-1
+              with:
+                  spec: project.yml
+                  quiet: false
+                  version: '2.41.0'
```

Comment:
I see this pattern in a few places, do you want me to refactor it using a GitHub composite action? I can contribute to your PR